apps/frontend/src/app/Data/Weapons/WeaponSheet.tsx

```diff
@@ -33,17 +34,14 @@ export default class WeaponSheet {
     this.sheet = weaponSheet
     this.data = data
   }
-  static trm(key: string) {
-    return (strKey: string) => <Translate ns={`weapon_${key}`} key18={strKey} />
+
+  static trm(key: WeaponKey) {
+    return trans('weapon', key)[1]
   }
-  static tr(key: string) {
-    return (strKey: string) => (
-      <Translate
-        ns={key === 'QuantumCatalyst' ? `weapon_${key}` : `weapon_${key}_gen`}
-        key18={strKey}
-      />
-    )
+  static tr(key: WeaponKey) {
+    return trans('weapon', key)[0]
   }
```

Comment:
There may be a more efficient way to write this, but I'm not that good with TypeScript